_posts/2021-10-12-devfile-v2-and-the-devworkspace-operator-p1.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Devfile v2 and the DevWorkspace Operator - Part 1
+layout: post
+author: Mario Loriedo
+description: >-
+  How to enable the DevWorkspace engine and what it means from a user point of view
+categories: []
+keywords: ['devfile', 'devworkpace']
+slug: /@mario.loriedo/devfile-v2-and-the-devworkspace-operator-p1
+---
+
+![Locally trusted certs]({{ site.url }}{{ site.baseurl }}/assets/img/devfile-v2-and-the-devworkspace-operator-part1/che-workspace-engines.png)
+
+With version [7.28 of Che](https://github.com/eclipse/che/releases/tag/7.28.0) we have introduced a new workspace engine, [the DevWorkspace Operator](https://github.com/devfile/devworkspace-operator), that supports v2 of the Devfile specification. Although the default engine is still the che-server we plan to replace it with the DevWorkspace in the next few months.
+
+Switching to the DevWorkspace engine has some important consequences. Notably on the authentication subsystem that will be lighter and more flexible, on the workspaces network managed by a central gateway powered by [Traefik](https://github.com/traefik/traefik) and simpler configuration options for Che administrators.  
```

Comment:
lighter is more or less true for OpenShift, but I would say not for Kubernetes (at least for local).
It makes sense to note that on Kubernetes now we require OpenID provide(or OAuth, I'm not sure)

But yeah, that's not something we already support. And on kubernetes we still depends on Keycloak.

Update. Ah, that's delayed till next blogpost:

> Keycloak is not required anymore and Che users have to be Kubernetes users